lib/controllers.js

```diff
@@ -464,32 +477,46 @@ module.exports = function(rootDirectory, logger, indexDocument, errorDocument) {
           }
           logger.info(
             'Stored object "%s" in bucket "%s" successfully',
-            req.params.key,
-            req.bucket.name
+            object.key,
+            object.bucket
           );
           triggerS3Event(req, res, {
-            bucket: req.bucket.name,
+            bucket: req.params.bucket,
             eventType: "Put",
-            S3Item: key
+            S3Item: new S3Object(object.bucket, object.key, {
+              "content-length": size,
+              etag: size
+            })
           });
-          res.header("ETag", JSON.stringify(key.md5));
+          res.header("ETag", JSON.stringify(md5));
           return res.status(200).end();
         });
       }
     },
 
+    genericPost(req, res) {
+      if (req.query.delete !== undefined) {
+        req.pipe(
+          concat(data => {
+            req.body = data;
+            deleteObjects(req, res);
+          })
+        );
+      } else {
+        notFoundResponse(req, res);
+      }
+    },
+
     postObject(req, res) {
+      let template;
```

Comment:
I know we follow this pattern elsewhere in the code, but there's no need for `template` to be mutable since we pass it into `buildXmlResponse` and return everywhere we set `template`. Instead, use `const template` right before calling `buildXmlResponse` to reduce the cognitive load of reading this code.